src/main/java/cubicchunks/world/ICubicWorld.java

```diff
@@ -147,6 +148,8 @@ default boolean testForCubeAndNeighbor(Cube cube, @Nullable Predicate<Cube> test
 	//field accessors
 	WorldProvider getProvider();
 
+	void hotSetProvider(WorldProvider provider);
```

Comment:
As I said before, keep it away from public interface. Or make it throw exception if it's too late to set the value.